internal/pkg/agent/application/managed_mode.go

```diff
@@ -259,14 +260,29 @@ func (m *managedConfigManager) wasUnenrolled() bool {
 	return false
 }
 
-func (m *managedConfigManager) initFleetServer(ctx context.Context) error {
-	ctx, cancel := context.WithTimeout(ctx, 15*time.Second)
+func (m *managedConfigManager) initFleetServer(ctx context.Context, cfg *configuration.FleetServerConfig) error {
+	fleetTimeoutEnvKey := "FLEET_TIMEOUT"
+	startTimeout := 30 * time.Second
+	// set timeout from config
+	if cfg != nil && cfg.InitTimeout != nil {
+		startTimeout = *cfg.InitTimeout
+	}
+	// override with env var, match the other FLEET_* settings
+	if envTimeout := os.Getenv(fleetTimeoutEnvKey); envTimeout != "" {
+		var err error
+		startTimeout, err = time.ParseDuration(envTimeout)
+		if err != nil {
+			return fmt.Errorf("error parsing %s value: %s: %w", fleetTimeoutEnvKey, envTimeout, err)
+		}
+	}
```

Comment:
You should change this to use a command-line argument `--fleet-server-timeout` so it can be used by the `install` and `enroll` command. 

Then you can have the environment variables for the containers to load this value here:

https://github.com/elastic/elastic-agent/blob/1546a773e7cf800a8256a0cb02379a96eeb0ec46/internal/pkg/agent/cmd/setup_config.go#L80